### PR TITLE
APP-2452: Fix confusing error on misformatted 'model' field

### PR DESCRIPTION
**Pull request by hexbabe**
https://viam.atlassian.net/browse/APP-2452

- Patched with in Benji's attached diff on Jira
- Same problem with cloud config still giving the clumsy error as before, but the theory is that if we deploy and app uses the latest version, the error will go away because it will invoke the new version of `UnmarshalJSON`

### CODE REVIEW

File: resource/model_test.go [hunk 1]
```diff
@@ -143,56 +143,56 @@ func TestModelFromString(t *testing.T) {
 			"acme/corp:test:modelA",
 			resource.Model{},
 			"not a valid model name",
-			"invalid character",
```
Comment (benjirewis) [inline resource/model_test.go:146]: For context, these expected errors were the ones returned by `json.Unmarshal` when it was unable to turn `arm/corp:test:modelA` into a model struct. Now, we return a more informative error if model doesn't match valid regexes and doesn't contain an opening brace.

**Review by benjirewis — APPROVED** [review]
LGTM assuming tests pass; thanks 🧑‍🔧

**Review by cheukt — APPROVED** [review]
LGTM, I think this still needs to be fixed on app tho (the error message is better but ideally it wouldn't be in logs (and instead the app should prevent users from saving something obviously broken)).

@hexbabe can you make a ticket on app for it?

**Comment by hexbabe** [comment]
Fleet follow-up ticket made: https://viam.atlassian.net/browse/APP-2885

**Pull request merged**